update aws amplify from v5 to v6 and fix its test

timheilman opened the pull request:
per https://github.com/cypress-io/cypress-documentation/pull/5610 , here are the updates applied to the cypress-real-world-app to fix the breaking changes in the amplify-js version 6 release

closes #1459

timheilman committed:
docs: fix typo in console error message for cognito index page

timheilman committed:
fix: global replace of case-sensitive VITE_APP with VITE;

this seems to have been a mistake during the upgrade from create-react-app to vite.  See historical commits 7b78b6eb6cb87ad22b4227f642ed4775f51c518b and e66f559f34a2dae163b366113363683f958f22c8 for the mystery of where VITE_APP came from.  REACT_APP should have been replaced with VITE but was replaced with VITE_APP instead.

timheilman committed:
chore: upgrade amplify-js library from version 5 to version 6 and

fix its example cypress tests for programmatic login and cy.origin() login.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/cypress-io/cypress-realworld-app?pullRequest=1513) <br/>All committers have signed the CLA.

timheilman committed:
docs: prettier fixes to README.md

AtofStryker commented on README.md:
```diff
@@ -221,7 +221,26 @@ A [guide has been written with detail around adapting the RWA](http://on.cypress
+      - openid
+  - The user pool must have an app client configured, with:
+    - enabled auth flow `ALLOW_USER_PASSWORD_AUTH`, only for programmatic login flavor of test.
+    - The cy.origin() flavor of test only requires auth flow `ALLOW_USER_SRP_AUTH`, and does not require `ALLOW_USER_PASSWORD_AUTH`.
```
```suggestion
    - The `cy.origin()` flavor of test only requires auth flow `ALLOW_USER_SRP_AUTH`, and does not require `ALLOW_USER_PASSWORD_AUTH`.
```

AtofStryker commented on README.md:
```diff
@@ -221,7 +221,26 @@ A [guide has been written with detail around adapting the RWA](http://on.cypress
+
+The test knobs are in a few places:
+
+- The .env file has `VITE_AUTH_TOKEN_NAME` and vars beginning `AWS_COGNITO`. Be careful not to commit any secrets.
```
```suggestion
- The `.env` file has `VITE_AUTH_TOKEN_NAME` and vars beginning `AWS_COGNITO`. Be careful not to commit any secrets.
```

AtofStryker commented on src/index.auth0.tsx:
```diff
@@ -22,14 +22,14 @@ const onRedirectCallback = (appState: any) => {
 
 /* istanbul ignore if */
-if (process.env.VITE_APP_AUTH0) {
+if (process.env.VITE_AUTH0) {
```
thank you for fixing/updating these!

timheilman committed:
build: update yarn lock changes only for amplify on chore branch

by: removing amplify from package.json, yarn install, replacing amplify in package.json, yarn install.

timheilman committed:
Merge branch 'develop' into chore/update-aws-amplify-tdh

fixed conflict by: git restore --staged yarn.lock ; git checkout yarn.lock ; git checkout develop -- yarn.lock ; PUPPETEER_SKIP_CHROMIUM_DOWNLOAD=true yarn install ; git add -u

timheilman committed:
Update README.md

backtick-quote cy.origin()

Co-authored-by: Bill Glesias <bglesias@gmail.com>

timheilman committed:
Update README.md

backtick-quote .env

Co-authored-by: Bill Glesias <bglesias@gmail.com>

timheilman committed:
Merge branch 'develop' into chore/update-aws-amplify-tdh

AtofStryker commented on src/containers/AppCognito.tsx:
```diff
@@ -39,35 +38,53 @@ const AppCognito: React.FC = /* istanbul ignore next */ () => {
 
       <PrivateRoutesContainer
-        isLoggedIn={isLoggedIn}
+        isLoggedIn={true}
```
is there a reason we just pass `true` into this or is this left over from development?

timheilman replied:
Ahh, this (I think) is a result of the change in the purpose of version 5 of the amplify-react-ui Authenticator component from what was done in version 1 of that library (which used an `AmplifyAuthenticator` component rather than `Authenticator`) and from what I suppose (though haven't looked closely) is done with okta, google, and auth0 in their tests presently.  The `AmplifyAuthenticator` I believe provided a login prompt for use _without redirection_, and the redirection would only occur once "Log in with AWS" was pressed within that`AmplifyAuthenticator`. This no-redirection scenario wasn't tested but if it were, would've been a direct login using ordinary authorization as coded in the express backend.

There was no similar login prompt provided by version 5 of amplify-react-ui, other than the `Authenticator` component, which is not designed to also provide a link to redirect to the hosted UI. `Authenticator`'s purpose is exclusively to do a non-redirect login.  So, since it wasn't tested anyway, I figured `AppCognito.tsx` would no longer offer that direct-login UI with `Authenticator` and instead redirect immediately to the hosted UI (line 57) upon page load, if unauthenticated.  The rendering of `AppCognito.tsx` is foregone until the redirect occurs (line 79).  As such, if `PrivateRoutesContainer` here is rendered at all, it is because `null` was not returned on line 79, because `isLoggedIn` is already true (post-redirect-and-redirect-back from the hosted UI).  `isLoggedIn` could still be passed here, but it will always be `true`.

AtofStryker replied:
probably just makes sense to just pass `true` since we do share this component with the other auth tests. That seems fine

timheilman commented on package.json:
```diff
@@ -13,7 +13,6 @@
   },
   "dependencies": {
     "@auth0/auth0-react": "2.2.4",
-    "@aws-amplify/ui-react": "^5.0.4",
```
Because `Authenticator` in `@aws-amplify/ui-react@5` changed from `AmplifyAuthenticator` in `@aws-amplify/ui-react@1`, and no longer provides a "Log In with AWS" link to perform an oauth-based redirect login using the hosted UI, this library is simply removed and redirection to the hosted UI occurs immediately upon load of `AppCognito.tsx` without needing to click "Log in with AWS" which had been provided by `AmplifyAuthenticator` back at version 1.

timheilman commented:
sorry for the re-review request. Fixing the `yarn.lock` conflict dismissed the prior review.
